src/posix/platform/radio.cpp

```diff
@@ -495,21 +488,13 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
 uint32_t otPlatRadioGetSupportedChannelMask(otInstance *aInstance)
 {
     OT_UNUSED_VARIABLE(aInstance);
-    return
-#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
-        sMaxPowerTable.GetAllowedChannelMask() &
-#endif
-        sRadioSpinel.GetRadioChannelMask(false);
+    return sRadioSpinel.GetRadioChannelMask(false);
```

Comment:
This changes the behavior, that is the supported channel can be dynamically changed as well. How about moving the max power table into radio spinel itself, so that the RCP restoration can use it as well?

Reply:
Done